Util/model_parser_cxx/model_parser.H

```diff
@@ -0,0 +1,307 @@
+                model::state(id-1, var_index)) /
+           (model::r(id) - model::r(id-1));
+       interp = slope * (r - model::r(id)) + model::state(id, var_index);
+
+
+       // safety check to make sure interp lies within the bounding points
+       Real minvar = amrex::min(model::state(id-1, var_index),
+                                model::state(id, var_index));
+       Real maxvar = amrex::max(model::state(id-1, var_index),
+                                model::state(id, var_index));
+       interp = amrex::max(interp, minvar);
+       interp = amrex::min(interp, maxvar);
+
+    } else {
+
+        if (r >= model::r(id)) {
+
+            slope = (model::state(id+1, var_index) -
+                     model::state(id, var_index)) /
+                (model::r(id+1) - model::r(id));
+            interp = slope * (r - model::r(id)) + model::state(id, var_index);
+
+        } else {
+
+            slope = (model::state(id, var_index) -
+                     model::state(id-1, var_index)) /
+                (model::r(id) - model::r(id-1));
+            interp = slope * (r - model::r(id)) + model::state(id, var_index);
+
+        }
+
+    }
+
+    return interp;
+
+}
+
+AMREX_INLINE
+void
+read_model_file(std::string& model_file) {
+
+    bool found_model, found_dens, found_temp, found_pres;
+    bool found_spec[NumSpec];
+
+    // read in the initial model
+
+    std::ifstream initial_model_file;
+
+    initial_model_file.open(model_file, std::ios::in);
+    if (!initial_model_file.is_open()) {
+        amrex::Error("Error opening the initial model");
+    }
+
+    std::string line;
+
+    // first the header line -- this tells us the number of points
+
+    getline(initial_model_file, line);
+    std::string npts_string = line.substr(line.find("=")+1, line.length());
+    model::npts = std::stoi(npts_string);
+
+    if (model::npts > NPTS_MODEL) {
+        amrex::Error("Error: model has more than NPTS_MODEL points,  Increase MAX_NPTS_MODEL");
+    }
+
+    // next line tells use the number of variables
+
+    getline(initial_model_file, line);
+    std::string num_vars_string = line.substr(line.find("=")+1, line.length());
+    int nvars_model_file = std::stoi(num_vars_string);
+
+    // now read in the names of the variables
+
+    std::vector<std::string> varnames_stored;
+    for (int n = 0; n < nvars_model_file; n++) {
+        getline(initial_model_file, line);
+        std::string var_string = line.substr(line.find("#")+1, line.length());
+        varnames_stored.push_back(model_string::ltrim(model_string::rtrim(var_string)));
+    }
+
+    // allocate storage for the model data
+
+    amrex::Print() << "reading initial model" << std::endl;
+    amrex::Print() << model::npts << " points found in the initial model" << std::endl;
+    amrex::Print() << nvars_model_file << " variables found in the initial model file" << std::endl;
+
+    // start reading in the data
+
+    Real vars_stored[nvars_model_file];
```

Comment:
Since variable-length arrays are not legal C++ this should probably be a vector.

Reply:
fixed